src/sentry/snuba/metrics/helpers.py

```diff
@@ -606,3 +608,59 @@ def translate_results(self):
         ]
 
         return groups
+
+
+def metric_object_factory(op, metric_name):
+    return RawMetric(op, metric_name)
+
+
+class MetricsFieldBase(ABC):
+    def __init__(self, op, metric_name):
+        self.op = op
+        self.metric_name = metric_name
+
+    def get_entity(self, **kwargs):
+        raise NotImplementedError
+
+    def generate_metric_ids(self, *args):
+        raise NotImplementedError
+
+    def generate_select_statements(self, **kwargs):
+        raise NotImplementedError
+
+    def generate_orderby_clause(self, **kwargs):
+        raise NotImplementedError
+
+
+class RawMetric(MetricsFieldBase):
+    def get_entity(self, **kwargs):
+        return OPERATIONS_TO_ENTITY[self.op]
+
+    def generate_metric_ids(self, entity, *args):
+        return (
+            {resolve_weak(self.metric_name)} if OPERATIONS_TO_ENTITY[self.op] == entity else set()
+        )
+
+    def _build_conditional_aggregate_for_metric(self, entity):
+        snuba_function = OP_TO_SNUBA_FUNCTION[entity][self.op]
+        return Function(
+            snuba_function,
+            [
+                Column("value"),
+                Function("equals", [Column("metric_id"), resolve_weak(self.metric_name)]),
+            ],
+            alias=f"{self.op}({self.metric_name})",
+        )
+
+    def generate_select_statements(self, entity, **kwargs):
+        return [self._build_conditional_aggregate_for_metric(entity=entity)]
+
+    def generate_orderby_clause(self, entity, direction, **kwargs):
+        return [
+            OrderBy(
+                self.generate_select_statements(entity=entity)[0],
+                direction,
+            )
+        ]
+
+    entity = cached_property(get_entity)
```

Comment:
nit: Is this property used anywhere?

Reply:
hm wait yeah, why do we do this? If we think it's needed it should be in the ABC

Reply:
Yeah I will need it in the upcoming PRs
It cannot be in the ABC because if I put it in the ABC it takes after the not implemented get_entity func on ABC